runtime/lua/vscode-neovim/api.lua

```diff
@@ -361,4 +365,64 @@ do
   end
 end
 
+--------------------------
+------ status item  ------
+--------------------------
+
+do
+  local _items = {}
+  local status_items = setmetatable({}, {
+    __newindex = function(_, k, v)
+      _items[k] = v
+      local status = vim.tbl_values(_items)
+      status = vim.tbl_filter(function(i)
+        return i ~= ""
+      end, status)
+      M.action("refresh-status-items", { args = { status } })
```

Comment:
Doing rpc work on a field assignment is rather surprising. I would suggest avoiding that kind of OOP cleverness. It doesn't really gain anything and adds uncertainty to the entire API. And it defeats the main benefit of having a table representation of the data: it's no longer data, it's a "live object"....

Instead I would suggest something like a `set_status_...` function.